cirq-core/cirq/ops/three_qubit_gates_test.py

```diff
@@ -110,6 +110,11 @@ def test_unitary():
         ),
         atol=1e-8,
     )
+    np.testing.assert_allclose(
+        cirq.unitary(cirq.FREDKIN**-1),
+        cirq.linalg.map_eigenvalues(cirq.unitary(cirq.FREDKIN), lambda b: b**-1),
```

Comment:
I don't think we need the call to `map_eigevalues`:

```python
np.testing.assert_allclose(cirq.unitary(cirq.FREDKIN**-1), cirq.unitary(cirq.FREDKIN), 1e-8)
```

Mathematical reason is that the eigenvalues of FREDKIN are all $\pm 1$, so $\lambda=\lambda^{-1}$. Pythonic reason is that AFAICT we should have equality `cirq.FREDKIN**-1 == cirq.FREDKIN`, before we even call `cirq.unitary`. Perhaps that is a better (or additional) thing to assert (though perhaps not inside `test_unitary`, not sure what a better suited test is). Any potential extra assert is up to you.

Reply:
Yeah, I agree we should add a `assert cirq.FREDKIN ** -1  == cirq.FREDKIN`. The reason I used this approach for the test here is to assert that the unitary matrix itself is self inverse, and not the object. Self inverse nature of the unitary matrix can be figured out using the math, but here the goal was to test that math.

Reply:
The simplest way to do this is

```python
u = cirq.unitary(cirq.FREDKIN)
np.testing.assert_allclose(u @ u, np.eye(8))
```

Reply:
Simplified the test for self-inverse as recommended. Also, note that `assert cirq.FREDKIN ** -1  == cirq.FREDKIN` is already covered in my original PR under the equality test `eq.add_equality_group(cirq.CSWAP(a, b, c), cirq.FREDKIN(a, b, c), cirq.FREDKIN(a, b, c) ** -1)`